src/foam/nanos/test/SmartTest.java

```diff
@@ -0,0 +1,145 @@
+package foam.nanos.test;
+
+import foam.core.X;
+import foam.core.EmptyX;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.System.err;
+
+public class SmartTest extends Test {
+  public static final String TEST_METHOD_SUFFIX = "Test";
+  public static final String CLEANUP_METHOD_SUFFIX = "Cleanup";
+  public static final String SETUP_METHOD_SUFFIX = "Setup";
+
+  private Method globalSetupMethod;
```

Comment:
Generally, I prefer to use protected over private as it makes things easier to test and debug.

Reply:
ok I'll user protected.